File: Python/백준/Bronze/5086. 배수와 약수/배수와 약수.py
import sys

a, b = -1, -1

while True:
    a, b = map(int, sys.stdin.readline().split(" "))
    
    if a == 0 and b == 0:
        break
    
    is_factor = b % a == 0
    is_multiple = a % b == 0

    if is_factor:
        print("factor")

    elif is_multiple:
        print("multiple")

    else:
        print("neither")

File: 백준/Bronze/5086. 배수와 약수/배수와 약수.cc
#include <iostream>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int a, b;

    while (true) {
        cin >> a >> b;
        
        if (a == 0 && b == 0) {
            break;
        }
        
        bool isFactor = false;
        bool isMultiple = false;

        if (a % b == 0) {
            isMultiple = true;
        }

        if (b % a == 0) {
            isFactor = true;
        }

        if (isFactor && isMultiple) {
            cout << "factor" << "\n";
        } else if (!isFactor && !isMultiple) {
            cout << "neither" << "\n";
        } else {
            cout << (isFactor ? "factor" : "multiple") << "\n";
        }
    }

    return 0;
}
